src/nemos/basis/_basis_mixin.py

```diff
@@ -0,0 +1,260 @@
+"""Mixin classes for basis."""
+
+from __future__ import annotations
+
+import copy
+import inspect
+from typing import Optional, Tuple, Union
+
+import numpy as np
+from numpy.typing import ArrayLike
+
+from ..convolve import create_convolutional_predictor
+from ._transformer_basis import TransformerBasis
+
+
+class EvalBasisMixin:
+    """Mixin class for evaluational basis."""
+
+    def __init__(self, bounds: Optional[Tuple[float, float]] = None):
+        self.bounds = bounds
+
+    def _compute_features(self, *xi: ArrayLike):
+        """
+        Apply the basis transformation to the input data.
+
+        The basis evaluated at the samples, or :math:`b_i(*xi)`, where :math:`b_i` is a
+        basis element. xi[k] must be a one-dimensional array or a pynapple Tsd.
+
+        Parameters
+        ----------
+        *xi:
+            The input samples over which to apply the basis transformation. The samples can be passed
+            as multiple arguments, each representing a different dimension for multivariate inputs.
+
+        Returns
+        -------
+        :
+            A matrix with the transformed features. The basis evaluated at the samples,
+            or :math:`b_i(*xi)`, where :math:`b_i` is a basis element. xi[k] must be a one-dimensional array
+            or a pynapple Tsd.
+
+        """
+        return self._evaluate(*xi)
+
+    def _set_kernel(self) -> "EvalBasisMixin":
```

Comment:
I forget if I mentioned this elsewhere: should `kernel_` exist in the superclass, or just in the Conv mixin? it's confusing to have it here and always None

Reply:
kernel_ exists in the super class, set to none. set_kernel should be in every subclass of Basis since it simplify the call on composite bases. Do you think that the _set_kernel logic should be moved in Basis, and have the ConvMixin method overwrite it? 
I thought we were supposed to have non-overlapping method names (so that the initialization order would not matter)

Reply:
Okay after talking with Edoardo: for now, this will stay, but the plan is to soon write a `fit` or `setup` method which is responsible for preparing the basis to be used. For both eval and conv, this will prepare the dimension of input/output. For conv, it will additionally set the kernel. And then this `setup` method will be the one that gets used by the composite and transformer basis objects.